src/beast/beast/crypto/detail/sha_context.h

```diff
@@ -0,0 +1,322 @@
+//------------------------------------------------------------------------------
```

Comment:
What is the rationale for putting all of this implementation detail in a header?  It slows compile time.  It doesn't help execution time.